### DISCUSSION TITLE
walletconnect_flutter_v2: Metamask connection request popup does not open

### DISCUSSION THREAD

**Discussion by uukait** [WalletConnect/walletconnect-monorepo#3815] [Developer Support]
I have implemented a Wallet connection in my mobile application using [`walletconnect_flutter_v2`](https://pub.dev/packages/walletconnect_flutter_v2), but I am unable to get a good connection.

Specifically, the metamask wallet application opens, but the connection request popup does not open within the wallet.

<img width="477" alt="Screenshot 2023-10-23 at 17 45 52" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/60546319/da4622fc-d564-4aac-8dce-12e4294c05a2">

**[Development environment]**
- Flutter: 3.13.8
- Android Studio Device: Pixel 7 
- Metamask: 7.8.0
- walletconnect_flutter_v2: ^2.0.0

**[Code]**

```dart
import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';

class SignIn extends StatelessWidget {
  static Web3App? _walletConnect;
  static const String launchError = 'Metamask wallet not installed';
  static const String kFullChainId = 'eip155:80001';
  static String? _url;
  static SessionData? _sessionData;

  String get deepLinkUrl => 'metamask://wc?uri=$_url';

  Future<void> _initWalletConnect() async {
    _walletConnect = await Web3App.createInstance(
      projectId: 'MY_PROJECTID',
      metadata: const PairingMetadata(
        name: 'Flutter WalletConnect',
        description: 'Flutter WalletConnect Dapp Example',
        url: 'https://walletconnect.com/',
        icons: [
          'https://walletconnect.com/walletconnect-logo.png',
        ],
      ),
    );
  }

  Future<String?> connectWallet() async {
    if (_walletConnect == null) {
      await _initWalletConnect();
    }

    final ConnectResponse connectResponse = await _walletConnect!.connect(
      requiredNamespaces: {
        'eip155': const RequiredNamespace(
          chains: ['eip155:80001'],
          methods: [
            'eth_signTransaction',
            'eth_sendTransaction',
          ],
          events: [
            'chainChanged',
            'accountsChanged',
          ],
        ),
      },
    );

    final Uri? uri = connectResponse.uri;

    if (uri == null) {
      return null;
    }

    final String encodedUrl = Uri.encodeComponent('$uri');

    _url = encodedUrl;

    await launchUrlString(
      deepLinkUrl,
      mode: LaunchMode.externalApplication,
    );

    _sessionData = await connectResponse.session.future;  // ★ The process is not returning.

    final String account = NamespaceUtils.getAccount(
      _sessionData!.namespaces.values.first.accounts.first,
    );

    return account;
  }
```

- Referenced issue for implementation: https://github.com/WalletConnect/WalletConnectFlutterV2/issues/113

No error logs are shown. I have checked the code with debugPrint and it seems to be stuck at `await connectResponse.session.future;`.

**Comment by glitch-txs** [comment]
Hi, could you please open an issue in Flutter repo? Thanks! 🙌

**Reply by uukait** [reply]
Thanks for the reply ☺️
Just to confirm, which `Flutter repo` are you referring to?

- https://github.com/flutter/flutter
- https://github.com/WalletConnect/WalletConnectFlutterV2

**Reply by glitch-txs** [reply]
WalletConnect one, I'm usually the most active here but I don't know much about flutter, so there the maintainer of the repo can have a better look :)

**Reply by uukait** [reply]
Got it, thank you so much! I will open an issue in the WalletConnect repo.
